supportedBackends/google/batch/src/main/scala/cromwell/backend/google/batch/actors/GcpBatchAsyncBackendJobExecutionActor.scala

```diff
@@ -1028,49 +1082,197 @@ class GcpBatchAsyncBackendJobExecutionActor(override val standardParams: Standar
+      status <- pollStatus(workflowId, jobName, backendSingletonActor, initializationData.requestFactory)
+    } yield status
   }
 
   override def isTerminal(runStatus: RunStatus): Boolean =
     runStatus match {
-      case _: RunStatus.TerminalRunStatus =>
-        log.info(s"isTerminal match terminal run status with $runStatus")
-        true
-      case other =>
-        log.info(f"isTerminal match _ running with status $other")
-        false
+      case _: RunStatus.TerminalRunStatus => true
+      case _ => false
     }
 
   override def isDone(runStatus: RunStatus): Boolean =
     runStatus match {
-      case _: RunStatus.Succeeded =>
-        log.info("GCP batch job succeeded matched isDone")
-        true
-      case _: RunStatus.UnsuccessfulRunStatus =>
-        log.info("GCP batch job unsuccessful matched isDone")
-        false
+      case _: RunStatus.Success => true
+      case _: RunStatus.UnsuccessfulRunStatus => false
       case _ =>
-        log.info(s"did not match isDone: $runStatus")
         throw new RuntimeException(
-          s"Cromwell programmer blunder: isDone was called on an incomplete RunStatus ($runStatus)."
+          s"Cromwell programmer blunder: isSuccess was called on an incomplete RunStatus ($runStatus)."
         )
     }
 
   override def getTerminalEvents(runStatus: RunStatus): Seq[ExecutionEvent] =
     runStatus match {
-      case t: RunStatus.TerminalRunStatus =>
-        log.warning(s"Tried to get terminal events on a terminal status without events: $runStatus")
-        t.eventList
-
+      case successStatus: RunStatus.Success => successStatus.eventList
       case unknown =>
         throw new RuntimeException(s"handleExecutionSuccess not called with RunStatus.Success. Instead got $unknown")
     }
 
+  override def retryEvaluateOutputs(exception: Exception): Boolean =
+    exception match {
+      case aggregated: CromwellAggregatedException =>
+        aggregated.throwables.collectFirst { case s: SocketTimeoutException => s }.isDefined
+      case _ => false
+    }
+
+  private lazy val standardPaths = jobPaths.standardPaths
+
+  override def handleExecutionFailure(runStatus: RunStatus, returnCode: Option[Int]): Future[ExecutionHandle] = {
+
+    def generateBetterErrorMsg(runStatus: RunStatus.UnsuccessfulRunStatus, errorMsg: String): String =
+      if (
+        runStatus.errorCode.getCode.value() == BatchFailedPreConditionErrorCode
+        && errorMsg.contains("Execution failed")
+        && (errorMsg.contains("Localization") || errorMsg.contains("Delocalization"))
+      ) {
+        s"Please check the log file for more details: $gcpBatchLogPath."
+      }
+      // If error code 10, add some extra messaging to the server logging
+      else if (runStatus.errorCode.getCode.value() == BatchMysteriouslyCrashedErrorCode) {
+        jobLogger.info(s"Job Failed with Error Code 10 for a machine where Preemptible is set to $preemptible")
```

Comment:
Does Batch use the same range of error codes? The enum `com.google.cloud.batch.v1.JobStatus.State` only goes up to 6 it looks like.